accepted/0000-subscription-matching-in-public-clouds.md

```diff
@@ -0,0 +1,194 @@
+- Feature Name: Subscription matching in public clouds
+- Start Date: 2019-08-14
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC set the guides for supporting "Subscription Matching" on virtual instances created on Public Clouds like AWS, Azure or Google Compute Engine. Exploring and gatehering virtual instances from those Public Clouds enables "Subscription Matching" to properly match subscriptions for "1-2 Virtual Machines".
+
+# Motivation
+[motivation]: #motivation
+
+Currently the subscription matcher does not work for VMs that are running in public clouds. Our `virtual-host-gatherer` tool doesn't support exploring those public clouds, so there is no effective way to provide the needed information that makes subscription matcher be aware of the virtualization environments and therefore match the subscription and benefits from "1-2 Virtual Machine" licencing.
+
+This RFC defines the procedures to enabling "Subscriptions Matching" on different Public Clouds providers.
+
+Also provides implementation details for implementing the following modules for `virtual-host-gatherer`:
+
+- A generic (JSON-file based) module that enables Administrators to provide its own custom virtual instances definition.
+- A module for gathering virtual instances from Azure Cloud.
+- A module for gathering virtual instances from AWS.
+- A module for gathering virtual instances from Google Compute Engine.
+
+This way, the Administrators will be able to define and/or gather their virtualization environments in the public clouds and make the subscription matcher do a succesfull matching of the "1-2 Virtual Machine" subscriptions.
+
+# Detailed design
+[design]: #detailed-design
+
+Most of the implementation effort needed is about coding each one of the needed plugins for `virtual-host-gatherer`. On the Java side, we would need to adapt `VirtualHostManagerProcessor.java` and most probably also `VirtualInstanceManager.java` in a similar way that we do when processing a KUBERNETES virtual host, since we don't always know about amount of RAM, number of CPUs, etc from the Virtualization Host.
+
+
+## 1-2 Virtual Machine Licensing
+From [SUSE Terms & Conditions](https://www.suse.com/products/terms_and_conditions.pdf):
+
+> Up to 2 Virtual Machines running on the same Virtualization Host or Virtualization Environment or within the same Private Cloudaccount or **Public Cloud zone can be deployed with one "1-2 Sockets or 1-2 Virtual Machines" Subscription Offering**.
+
+This means, all virtual instances that belong to the same "zone" (not region) will match with the "1-2 Virtual Machines" subscription.
```

Comment:
If zone=! region, then please define zone

Reply:
What scares me: `Private Cloudaccount` **or** `Public Cloud zone`. Everything which is gathered with 1 account can be grouped together? @joesusecom ?

And I would also like to know what `zone` is.

Reply:
I will clarify differences between zone and region on the RFC text.

AFAIU `Private Cloud account` refers i.a. to a private VMware cloud (like we already do in our vCenter example). So you define an Account, and all VMs gatherer from VMware are grouped together.

For the Public Cloud, it explicitly says "Public Cloud zone" (and not account), so this would mean, we cannot group by account but instead by "zone". Therefore, a customer using i.a. AWS will need to create as many "Virtual Host Manager" on SUSE Manager as "zones" they want to gather. Then each one of those "Virtual Host Manager" are considered as independent virtualization groups for "1-2 subscriptions" matching.